spec/spec_helper.rb

```diff
@@ -5,15 +5,15 @@
 
 RSpec.configure do |config|
   # DatabaseCleaner
-  config.before(:suite) {
-    DatabaseCleaner.clean_with :deletion, except: ['spree_countries', 'spree_states']
-  }
-  config.before(:each)           { DatabaseCleaner.strategy = :transaction }
-  config.before(:each, concurrency: true) {
+  config.before(:each, concurrency: true) do
+    config.use_transactional_fixtures = false
     DatabaseCleaner.strategy = :deletion, { except: ['spree_countries', 'spree_states'] }
-  }
-  config.before(:each)           { DatabaseCleaner.start }
-  config.after(:each)            { DatabaseCleaner.clean }
```

Comment:
Rails uses transactions by default and the rollback at the end is all that's needed for cleaning the database. We were already using that in system specs. But for historical reasons we used DatabaseCleaner instead of Rails for the transactions before. So I thought that we can just use the Rails default where it works and use the DatabaseCleaner only in cases where we need it, which is only specs that start threads to test race conditions. Wrapping those in a transaction results in the threads having isolated transactions and can't see each other's data.